### DISCUSSION TITLE
customManager with currentVersion specified twice

### DISCUSSION THREAD

**Discussion by morremeyer** [renovatebot/renovate#25590] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want to update the SheetJS node dependency with renovate. This general process is documented in https://docs.sheetjs.com/docs/getting-started/installation/nodejs/.

As the documentation specifies, the .tgz file is at an URL that looks as follows:

```
https://cdn.sheetjs.com/xlsx-0.20.0/xlsx-0.20.0.tgz
```

This cannot be matched by a working regex, since it has `currentVersion` specified twice. 
Is there a way to get this working with a valid regex?

### What you tried so far.

This can be checked in https://github.com/envelope-zero/you-need-a-parser/pull/190.

I tried using 

```json
{
  "customManagers": [
    {
      "description": "SheetJS",
      "matchStrings": [
        "https://cdn.sheetjs.com/xlsx-(?<currentValue>.*)/xlsx-(?<currentValue>.*)\\.tgz"
      ],
      "datasourceTemplate": "gitea-tags",
      "registryUrlTemplate": "https://git.sheetjs.com",
      "depNameTemplate": "sheetjs/sheetjs",
      "fileMatch": ["packages/ynap-parsers/package.json"]
    }
  ]
}
```

This fails the renovate configuration validation with

```json
        "errors": [
         {
           "topic": "Configuration Error",
           "message": "Invalid regExp for customManagers: `https://cdn.sheetjs.com/xlsx-(?<currentValue>.*)/xlsx-(?<currentValue>.*)\\.tgz`"
         }
       ]
```

See for example https://github.com/envelope-zero/you-need-a-parser/actions/runs/6762938053/job/18379476894.

**Comment by rarkins** [comment]
If this were a "first class" manager of its own, we'd probably treat the second part of the filename as part of "artifacts" and sync it after the fact.

I wonder if we could find a middle ground for custom managers which would allow you to denote another part of the string as artifacts text and then Renovate would search-replace all version strings within it

**Comment by viceice** [comment, marked as answer]
did you tried to only capture the first version as current value? I think renovate regex manager will replace all versions inside match string. 🤔

**Reply by morremeyer** [reply]
I never merged these changes to the default branch since the validation already failed. I can try merging it, and if it works we'd just have to fix the schema validation instead of the behavior.

**Reply by viceice** [reply]
no, don't merge as is, remove the group name from the second version match before. that should fix the validation

**Reply by morremeyer** [reply]
Like this?

```
"https://cdn.sheetjs.com/xlsx-(?<currentValue>.*)/xlsx-(.*)\\.tgz"
```

**Reply by viceice** [reply]
yes

**Reply by rarkins** [reply]
I know we _used_ to replace all, but I thought we changed that (it used to occasionally be undesirable too, e.g. if the version was something very common like `1`). I guess we'll soon see!

**Reply by morremeyer** [reply]
It still works: [envelope-zero/you-need-a-parser@`7a2d6b1` (#194)](https://github.com/envelope-zero/you-need-a-parser/pull/194/commits/7a2d6b1e6a537ce207ae86796b05da81dc52275f). 

I didn't see this specifically mentioned in the docs, it might be a good addition.
